src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceLoggerExtension.cs

```diff
@@ -0,0 +1,123 @@
+
+        private static readonly EventId _scriptHostServiceInitCanceledByRuntime = new EventId(0, $"{_codeArea}\\{nameof(ScriptHostServiceInitCanceledByRuntime)}");
+        private static readonly EventId _unehealthyCountExceeded = new EventId(0, $"{_codeArea}\\{nameof(UnhealthyCountExceeded)}");
+        private static readonly EventId _offline = new EventId(0, $"{_codeArea}\\{nameof(Offline)}");
+        private static readonly EventId _initializing = new EventId(0, $"{_codeArea}\\{nameof(Initializing)}");
+        private static readonly EventId _initialization = new EventId(0, $"{_codeArea}\\{nameof(Initialization)}");
+        private static readonly EventId _inStandByMode = new EventId(0, $"{_codeArea}\\{nameof(InStandByMode)}");
+        private static readonly EventId _unhealthyRestart = new EventId(0, $"{_codeArea}\\{nameof(UnhealthyRestart)}");
+        private static readonly EventId _stopping = new EventId(0, $"{_codeArea}\\{nameof(Stopping)}");
+        private static readonly EventId _didNotShutDown = new EventId(0, $"{_codeArea}\\{nameof(DidNotShutDown)}");
+        private static readonly EventId _shutDownCompleted = new EventId(0, $"{_codeArea}\\{nameof(ShutDownCompleted)}");
+        private static readonly EventId _skipRestart = new EventId(0, $"{_codeArea}\\{nameof(SkipRestart)}");
+        private static readonly EventId _restarting = new EventId(0, $"{_codeArea}\\{nameof(Restarting)}");
+        private static readonly EventId _restarted = new EventId(0, $"{_codeArea}\\{nameof(Restarted)}");
+        private static readonly EventId _building = new EventId(0, $"{_codeArea}\\{nameof(Building)}");
+        private static readonly EventId _startupWasCanceled = new EventId(0, $"{_codeArea}\\{nameof(StartupWasCanceled)}");
+        private static readonly EventId _errorOccured = new EventId(0, $"{_codeArea}\\{nameof(ErrorOccured)}");
+        private static readonly EventId _errorOccuredInactive = new EventId(0, $"{_codeArea}\\{nameof(ErrorOccuredInactive)}");
+        private static readonly EventId _cancellationRequested = new EventId(0, $"{_codeArea}\\{nameof(CancellationRequested)}");
+
+        public static void ScriptHostServiceInitCanceledByRuntime(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _scriptHostServiceInitCanceledByRuntime, "Initialization cancellation requested by runtime.");
+        }
+
+        public static void UnhealthyCountExceeded(this ILogger logger, int healthCheckThreshold, TimeSpan healthCheckWindow)
+        {
+            logger.Log(LogLevel.Error, _unehealthyCountExceeded, $"Host unhealthy count exceeds the threshold of {healthCheckThreshold} for time window {healthCheckWindow}. Initiating shutdown.");
+        }
+
+        public static void Offline(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _offline, "Host is offline.");
+        }
+
+        public static void Initializing(this ILogger logger)
```

Comment:
To confirm, the pattern you're establishing here is to use EventIds only for log statements we decide are important that we want to be able to query on, or have detectors be able to identify? I.e. we're not planning on converting all logging to this.

Reply:
We are going is to cover whole code base step by step. It is the first part. Eventually we should not have direct calls _looger.Log(...).